File: 2014/2014Robot/Subsystems/CompressorSubsystem.cpp
#include "CompressorSubsystem.h"
#include "../Robotmap.h"
#include "../Commands/RunCompressorCommand.h"

CompressorSubsystem::CompressorSubsystem() : Subsystem("CompressorSubsystem") {
	printf("[CompressorSubsystem] Starting Construction\n");
	this->compressor = new Compressor(PRESSURE_SWITCH_CHANNEL, COMPRESSOR_RELAY_CHANNEL);
	printf("[CompressorSubsystem] Finished Construction\n");
}
    
void CompressorSubsystem::InitDefaultCommand() {
	// Set the default command for a subsystem here.
	SetDefaultCommand(new RunCompressorCommand());
}
 void CompressorSubsystem::Start() {
	 printf("[CompressorSubsystem] Starting Compressor\n");
	 this->compressor->Start();
 }
 
 void CompressorSubsystem::Stop() {
	 printf("[CompressorSubsystem] Stopping Compressor\n");
	 this->compressor->Stop();
 }

 bool CompressorSubsystem::GetPressureSwitch() {
	 return this->compressor->GetPressureSwitchValue() == 1;
 }

File: 2016/DriveBot/src/com/kennedyrobotics/robot2016/subsystems/CompressorSubsystem.java
package com.kennedyrobotics.robot2016.subsystems;

import com.kennedyrobotics.robot2016.commands.ComrpressorSupervisorCommand;

import edu.wpi.first.wpilibj.Compressor;
import edu.wpi.first.wpilibj.command.Subsystem;

/**
 *
 */
public class CompressorSubsystem extends Subsystem {
    
	private Compressor compressor = new Compressor();
	
    // Put methods for controlling this subsystem
    // here. Call these from Commands.

    public void initDefaultCommand() {
        // Set the default command for a subsystem here.
        //setDefaultCommand(new MySpecialCommand());
    	setDefaultCommand(new ComrpressorSupervisorCommand());
    }
    
    public void start() {
    	compressor.start();    	
    }
    
    public void stop() {
    	compressor.stop();
    }
       
}

